src/main/scala/com/rawlabs/das/server/cache/iterator/ExpressionEvaluator.scala

```
@@ -281,7 +281,7 @@ object ExpressionEvaluator {
           evalEquals(x, y)
         }
 
-      case _ => false
+      case _ => true
```

Predicates that aren't implemented are considered as `true` since they need to be evaluated in Postgres.

---

I don't agree with this part. (And in any case, this comment would have belonged in the source code, not in the PR).

But this is an ExpressionEvaluator component, so I'd expect it to act as one. This has nothing to do with Postgres, as this is its own standalone component, which is in no way depending or relating to how Postgres or FDW happen to use it.

If the predicates are not implemented, we either say we don't support it (return an option, exception, etc) or something else along those lines. In the interaction point between components, then we can assume that not being implemented provides an explicit semantic; but that's for the composition component/layer to do, not the implementation. Providing a semantic as this one here is crosscutting two components and concerns that are not related to each other.

---

If patched the ExpressionEvaluator to throw when an expression isn't supported. The layer above, QualEvaluator, is the one who logs the warning, ignores the failure and interprets the predicate as true (meaning it ignores the predicate as if it's not supported). It's also the one that interprets null as false. The QualEvaluator was changed to evaluate predicates one by one instead of as an "and", so that one can skip failing ones.